selfdrive/car/hyundai/carcontroller.py

```diff
@@ -88,20 +88,29 @@ def update(self, CC, CS):
           self.last_resume_frame = self.frame
 
     if self.frame % 2 == 0 and self.CP.openpilotLongitudinalControl:
-      lead_visible = False
+      pid_control = (actuators.longControlState == LongCtrlState.pid)
+      stopping = (actuators.longControlState == LongCtrlState.stopping)
       accel = actuators.accel if CC.longActive else 0
+      # TODO: aEgo lags when jerk is high, use smoothed ACCEL_REF_ACC instead?
+      accel_error = accel - CS.out.aEgo if CC.longActive else 0
 
-      jerk = clip(2.0 * (accel - CS.out.aEgo), -12.7, 12.7)
+      # TODO: jerk upper would probably be better from longitudinal planner desired jerk?
+      jerk_upper = clip(2.0 * accel_error, 0.0, 1.5) # zero when error is negative to keep decel control tight
```

Comment:
the positive jerk clip at 1.5 isn't ideal here, but 12.7 seems crazy high to me, so what should we clip it to?

Reply:
ISO 15622:2018 has no limits for positive jerk. Guess you can never reach something that high anyway on an ICE car.

Reply:
That makes sense, but since acceleration is going to always have bad lag at times on gas cars, which leads to big error, which leads to huge jerk allowed, I think it makes sense to clip this at something reasonable to keep driving comfortable and ensure massive overshoot doesn't happen.  Harald and I talked and we think 2 m/s3 is the most we ever really want once you are moving (this PR handles setting positive jerk to initially get moving separately) so I raised it to that.